### PR TITLE
VSCode DevContainers bootstrapping cleanup

### PR DESCRIPTION
**Pull request by msheiny**
# Closes: None
# What's Changed

* Removes some documentation that was no longer relevant related to VSCode DevContainers. Specifically, there is no need  to create a `docker-compose.override.yml` to pass in the ARCH on Mac Silicon machines. This was fixed on the `Dockerfile` itself using [a global TARGETARCH variable at some point](https://github.com/nautobot/nautobot/pull/4906).
* Few couple helper `invoke` lines to generate an `.env` docker compose file for building. Not critical need but negates the need to keep this static default file up-to-date.
* Updates the `type` in the vscode workspace file from `python` -> `debugpy` (at some point they'll deprecate the `python` type completely according to their documentation. I just noticed VSCode was flagging it while I was inspecting the file.
* Adds the `development/docker-compose.vscode-rdb.yml` file to the DevContainer build file.

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments)) - Couldn't grok whether this was needed without a previous issue attached?
- [ ] ~Attached Screenshots, Payload Example~ N/A
- [ ] ~Unit, Integration Tests~ Not needed, purely dev environment changes
- [x] Documentation Updates (when adding/changing features)
- [ ] ~Example App Updates (when adding/changing features)~ N/A
- [ ] ~Outline Remaining Work, Constraints from Design~ N/A

### CODE REVIEW

**Review by glennmatthews — APPROVED** [review]
Thank you! I don't use the VSCode dev container myself but these changes look reasonable to me so I'll take your word for it :-)

**Comment by glennmatthews** [comment]
> Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments)) - Couldn't grok whether this was needed without a previous issue attached?

The CI tooling does enforce such a file regardless, but for a PR with no prior issue, you can use the PR number instead. Thanks!

**Comment by msheiny** [comment]
> > Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments)) - Couldn't grok whether this was needed without a previous issue attached?
> 
> The CI tooling does enforce such a file regardless, but for a PR with no prior issue, you can use the PR number instead (so in this case, probably `changes/6688.housekeeping` would be appropriate). Thanks!

okay, added to the latest revision 👍

File: tasks.py [hunk 1]
```diff
@@ -465,6 +465,12 @@ def vscode(context):
     """Launch Visual Studio Code with the appropriate Environment variables to run in a container."""
     command = "code nautobot.code-workspace"
 
+    # Setup PYTHON
+    env_file_path = os.path.join(BASE_DIR, "development/.env")
+    if not os.path.exists(env_file_path):
+        with open(env_file_path, "w") as env_file_obj:
+            env_file_obj.write(f"PYTHON_VER={context.nautobot.python_ver}")
+
     context.run(command, env={"PYTHON_VER": context.nautobot.python_ver})
```
Comment (glennmatthews) [inline tasks.py:474]: I'm curious - after #6659, do we need to also pass `NAUTOBOT_VER` to either of these calls, or is it not needed?

Comment (msheiny) [reply tasks.py:474]: Ahh this call is just running `code nautobot.code-workspace` to reopen the current session using the workspace file so I dont really think we need `PYTHON_VER` passed into that at all. It's not like it keeps that environment around in my VSCode session (it doesn't seem to be visible in my terminal session at least)

**Pull request merged**